src/createrepo_c.c

```diff
@@ -578,6 +578,66 @@ static int strlensort(gconstpointer a, gconstpointer b)
     }
 }
 
+// Sorting function for DuplicateLocation pointers, by pkg build time.
+// Compatible with g_array_sort()
+static int buildtimesort(gconstpointer a, gconstpointer b)
+{
+    // Function is supposed to take a double-pointer so unfortunately you cannot pass a
+    // string-comparison function directly.
+    struct DuplicateLocation *a_loc = (struct DuplicateLocation *)a;
+    struct DuplicateLocation *b_loc = (struct DuplicateLocation *)b;
+
+    assert(a_loc->pkg->time_build != 0);
+    assert(b_loc->pkg->time_build != 0);
+
+    // order by build time first
+    int64_t result = a_loc->pkg->time_build - b_loc->pkg->time_build;
+    if (result)
+        return result;
+
+    // and then alphabetically by the rpm location
+    return g_strcmp0(a_loc->location, b_loc->location);
+}
+
+
+static int
+handle_nevra_duplicates(GArray *locations, CmdDupNevra option)
+{
+    int skipped = 0;
+    for (int i=0; i<locations->len; i++) {
```

Comment:
Here we compare signed `int i` with unsigned `locations->len`,
I know its already present elsewhere but these things show up in covscans and other analyzers.
Can you change the `i` to unsigned? To `size_t` perhaps?